contacts/urls.py

```diff
@@ -1,18 +1,19 @@
 from django.conf.urls import url
 from django.contrib.auth.decorators import login_required
 
+from contacts.decorators import user_added_contact
 from . import views
 
 
 app_name = 'contacts'
 urlpatterns = [
     url(r'^contact/?$', login_required(views.ContactsListView.as_view()), name='contact_list'),
-    url(r'^(?P<pk>[0-9]+)/?$', login_required(views.ContactDetailView.as_view()), name='contact_detail'),
+    url(r'^(?P<pk>[0-9]+)/?$', login_required(user_added_contact(views.ContactDetailView.as_view())), name='contact_detail'),
     url(r'^add_contact/?$', login_required(views.AddContactView.as_view()), name='add_contact'),
     url(r'^add_address/?$', login_required(views.AddContactAddressView.as_view()), name='add_address'),
-    url(r'^update_contact/(?P<pk>[0-9]+)/?$', login_required(views.UpdateContactView.as_view()), name='update_contact'),
+    url(r'^update_contact/(?P<pk>[0-9]+)/?$', login_required(user_added_contact(views.UpdateContactView.as_view())), name='update_contact'),
     url(r'^update_contact_address/(?P<pk>[0-9]+)/?$',
-        login_required(views.UpdateContactAddressView.as_view()), name='update_contact_address'
+        login_required(user_added_contact(views.UpdateContactAddressView.as_view())), name='update_contact_address'
```

Comment:
in this url, pk means pk of Address, not Contact!
we need to update our view and decorator too to handle addresses as well as contacts.

Reply:
@MaryKhan 
I don't think this is even required. To add address against a contact, there is a drop down of added contacts which is displayed and the contacts are user specific. Let me know if you still think this should be done.

Reply:
what we do on updating an address?

Reply:
@MaryKhan 
Only the user who added the contact can add/edit contact's address too. Otherwise it would display a 403 Forbidden

Reply:
i don't think its working like that!
because in decorator, we're querying on Contact model (with pk), but in `UpdateContactAddressView`, pk would Address' pk.